PINCache/PINCache.m

```diff
@@ -152,157 +141,85 @@ - (void)setObject:(id <NSCoding>)object forKey:(NSString *)key block:(PINCacheOb
 {
     if (!key || !object)
         return;
+  
+    PINOperationGroup *group = [PINOperationGroup asyncOperationGroupWithQueue:self.operationQueue];
     
-    dispatch_group_t group = nil;
-    PINMemoryCacheObjectBlock memBlock = nil;
-    PINDiskCacheObjectBlock diskBlock = nil;
-    
-    if (block) {
-        group = dispatch_group_create();
-        dispatch_group_enter(group);
-        dispatch_group_enter(group);
-        
-        memBlock = ^(PINMemoryCache *memoryCache, NSString *memoryCacheKey, id memoryCacheObject) {
-            dispatch_group_leave(group);
-        };
-        
-        diskBlock = ^(PINDiskCache *diskCache, NSString *diskCacheKey, id <NSCoding> memoryCacheObject) {
-            dispatch_group_leave(group);
-        };
-    }
-    
-    [_memoryCache setObject:object forKey:key block:memBlock];
-    [_diskCache setObject:object forKey:key block:diskBlock];
+    [group addOperation:^{
```

Comment:
The idea is that they should be parallelized. I also want to be able cancel them (in the future) and separating allows that to be more likely to occur.